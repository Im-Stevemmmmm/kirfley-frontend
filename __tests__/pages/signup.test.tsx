import { gql } from '@apollo/client';
import { MockedProvider } from '@apollo/client/testing';
import { act, render } from '@testing-library/react';
import ReactDOM from 'react-dom';
import Signup from '../../pages/signup';

const REGISTER_USER = gql`
  mutation RegisterUser(
    $username: String!
    $email: String!
    $password: String!
  ) {
    registerUser(
      options: { username: $username, email: $email, password: $password }
    ) {
      user {
        username
        email
        password
      }
    }
  }
`;

const CHECK_FIELD_AVAILABILITY = gql`
  query CheckFieldAvailabilty($field: String!, $value: String!) {
    checkFieldAvailability(field: $field, value: $value) {
      successful
    }
  }
`;

const mocks = [
  {
    request: {
      query: REGISTER_USER,
      variables: {
        username: 'steve',
        email: 'example@email.com',
        password: '1234',
      },
    },
    result: {
      data: {
        registerUser: {
          user: {
            username: 'steve',
            email: 'example@email.com',
            password: '1234',
          },
        },
      },
    },
  },
  {
    request: {
      query: CHECK_FIELD_AVAILABILITY,
      variables: {
        field: 'username',
        value: 'steve',
      },
    },
    result: {
      data: {
        checkFieldAvailability: {
          successful: false,
        },
      },
    },
  },
  {
    request: {
      query: CHECK_FIELD_AVAILABILITY,
      variables: {
        field: 'email',
        value: 'example@email.com',
      },
    },
    result: {
      data: {
        checkFieldAvailability: {
          successful: true,
        },
      },
    },
  },
];

describe('sign up form', () => {
  it('renders properly', () => {
    const container = document.createElement('div');

    act(() => {
      ReactDOM.render(
        <MockedProvider mocks={mocks} addTypename={false}>
          <Signup />
        </MockedProvider>,
        container
      );
    });

    expect(container).toMatchSnapshot();
  });

  describe('queries', () => {
    beforeEach(() => {
      render(
        <MockedProvider mocks={mocks} addTypename={false}>
          <Signup />
        </MockedProvider>
      );
    });

    it('does not submit if the email or username is already registered', () => {});

    // it('signs a user up on submit', async () => {
    //   const submitButton = screen.getByText('Sign Up', {
    //     selector: 'button',
    //   });

    //   const username = screen.getByText('Username');
    //   username.textContent = 'steve';

    //   const email = screen.getByText('Email');
    //   email.textContent = 'example@email.com';

    //   const password = screen.getByText('Password');
    //   password.textContent = '1234';

    //   fireEvent.click(submitButton);
    // });
  });
});
