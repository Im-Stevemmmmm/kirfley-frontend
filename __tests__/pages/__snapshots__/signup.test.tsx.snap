// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sign up form renders properly 1`] = `
<div>
  <div
    id="root"
  >
    <div
      id="leftColumn"
    >
      <form>
        <h1>
          Sign Up
        </h1>
        <label
          for="username"
        >
          Username
        </label>
        <input
          id="username"
          name="username"
          placeholder="atleast 4 characters long"
          type="text"
          value=""
        />
        <label
          for="email"
        >
          Email
        </label>
        <input
          id="email"
          name="email"
          placeholder="example@domain.com"
          type="email"
          value=""
        />
        <label
          for="password"
        >
          Password
        </label>
        <input
          id="password"
          name="password"
          placeholder="atleast 4 characters long"
          type="password"
          value=""
        />
        <label
          for="confirmPassword"
        >
          Confirm Password
        </label>
        <input
          id="confirmPassword"
          name="confirmPassword"
          type="password"
          value=""
        />
        <button
          type="submit"
        >
          Sign Up
        </button>
      </form>
      <div
        id="background"
      />
    </div>
    <div
      id="rightColumn"
    />
  </div>
</div>
`;
